src/services/refactors/moveToFile.ts

```diff
@@ -915,26 +912,91 @@ function getRangeToMove(context: RefactorContext): RangeToMove | undefined {
+    /**[|const a = 1;
+       |]const b = 2;
+     */
+    if (afterEndNodeIndex !== -1 && range.end <= statements[afterEndNodeIndex].getStart()) {
+        afterEndNodeIndex = afterEndNodeIndex - 1;
+    }
+
+    const endingOverloadRangeToMove = getOverloadRangeToMove(file, statements[afterEndNodeIndex]);
+
+    /**[|function add1(x: number, y: number): number;
+       function add1(x: string, y: string): string;
+       function foo() { }
+       const a = 1;
+       function add2(x: boolean): string;
+       function add2(x: number): number;|]
+     */
+    if (endingOverloadRangeToMove && overloadRangeToMove) {
+        return {
+            toMove: statements.slice(findIndex(statements, s => s.pos === overloadRangeToMove.toMove[0].pos), endingOverloadRangeToMove.end + 1),
+            afterLast: endingOverloadRangeToMove.end === statements.length - 1 ? undefined : statements[endingOverloadRangeToMove.end + 1]
+        };
     }
 
-    // Can't only partially include the start node or be partially into the next node
-    if (range.pos > startStatement.getStart(file)) return undefined;
-    const afterEndNodeIndex = findIndex(statements, s => s.end > range.end, startNodeIndex);
-    // Can't be partially into the next node
-    if (afterEndNodeIndex !== -1 && (afterEndNodeIndex === 0 || statements[afterEndNodeIndex].getStart(file) < range.end)) return undefined;
+    /**function [|add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       function foo() { }|]
+                        or
+       function [|add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       function foo() { }
+       |]
+                        or
+       [|function add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       function foo() { }
+       |]const a = 1;
+     */
+    if (overloadRangeToMove) {
+        return {
+            toMove: statements.slice(findIndex(statements, s => s.end === overloadRangeToMove.toMove[0].end), afterEndNodeIndex === -1 ? statements.length : afterEndNodeIndex + 1),
+            afterLast: afterEndNodeIndex === -1 || afterEndNodeIndex === statements.length - 1 ? undefined : statements[afterEndNodeIndex + 1],
+        };
+    }
 
+    /**[|const a = 1;
+     * function add(x: number, y: number): number;
+       function add(x: string, y: string): string;|]
+                        or
+       [|
+       function add(x: number, y: number): number;
+       function add(x: string, y: string): string;
+       |]
+                        or
+       const a = 1;[|
+       function add(x: number, y: number): number;
+       function add(x: string, y: string): string;|]
+     */
+    if (endingOverloadRangeToMove) {
+        return {
+            toMove: statements.slice(startNodeIndex, endingOverloadRangeToMove.end === statements.length - 1 ? statements.length : endingOverloadRangeToMove.end + 1),

```

Comment:
There are a lot of intermediates here. Be aggressive about lifting them into their own variables so that it's more obvious what each one is for and to cut down on the line length.

Reply:
Does this look okay?